src/components/AddEditMovie/AddEditMovie.jsx

```diff
@@ -1,93 +1,171 @@
 import React from 'react'
 import "./add_edit_movie.css"
+import { isValidUrl, isNumber, isValidDate } from '../../utils/Validation'
+import { FilterCategories } from '../../utils/Constants'
+import { useFormik } from "formik"
 
 export default function AddEditMovie(props) {
+    const movieFilterArray = FilterCategories();
+    let isSubmitButton = false;
+    var boxRef = React.createRef();
 
-    var addMovieJson = {};
-
-    if (props.editMovieSelection !== undefined) {
-        addMovieJson = props.editMovieSelection;
-    }
-    else {
-        addMovieJson.src = "./src/resources/movie_poster2.png" //Temperory to refer to internal asset location
-    }
-
-    function handleTitleChange(e) {
-        addMovieJson.title = e.target.value;
-    }
-
-    function handleDateChange(e) {
-        addMovieJson.year = e.target.value;
-    }
-
-    function handleUrlChange(e) {
-        addMovieJson.src = e.target.value;
-    }
-
-    function handleGenreChange(e) {
-        addMovieJson.description = e.target.value;
-    }
-
-    function handleOverviewChange(e) {
-        addMovieJson.overview = e.target.value;
-    }
-
-    function handleRuntimeChange(e) {
-        addMovieJson.runtime = e.target.value;
+    function submitRequest() {
+        isSubmitButton = true;
+        formik.handleSubmit();
     }
 
-    let resetFields = () => {
-        if (props.editMovieSelection !== undefined) {
-            document.getElementById("form-edit-movie").reset()
-        }
-        else {
-            document.getElementById("form-add-movie").reset()
-        }
-    }
+    const formik = (props.editMovieSelection !== undefined) ?
+        useFormik({
+            initialValues: props.editMovieSelection,
```

Comment:
Looks like you have the same validation rules for both types of forms. You can move this validation to the separate function. Also, you can create **props.editMovieSelection** as an empty object for AddMovie form using [Default Props](https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values) functionality.